src/App.js

```diff
@@ -30,39 +30,109 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+
+  const swapPlayers = () => {
+    if (player === PLAYER_1) {
+      setPlayer(PLAYER_2)
+    }
+    if (player === PLAYER_2){
+      setPlayer(PLAYER_1)
+    }
+    };
+
+    const checkForWinner = () => {
+      // Complete in Wave 3
+      // You will need to:
+      // 1. Go across each row to see if 
+      //    3 squares in the same row match
+      //    i.e. same value
+      // 2. Go down each column to see if
+      //    3 squares in each column match
+      // 3. Go across each diagonal to see if 
+      //    all three squares have the same value.
+      
+      
+      // for each row, 
+      // if row[0] === X && row[1] === X && row[2] === X, winner = X
+      //
+      // rowcheck, column check, diaganol check 
+      // row check for in loop inside of for each
+      // col check for loop outside of for each 
+
+      for(let i in squares){
+        if (squares[i][0].value === squares[i][1].value && squares[i][1].value === squares[i][2].value &&  squares[i][0].value !== ''){
+          return `Winner is player ${squares[i][0].value}`;
+        }
+        else if(squares[0][i].value === squares[1][i].value && squares[1][i].value === squares[2][i].value &&  squares[0][i].value !== ''){
+          return `Winner is player ${squares[0][i].value}`;
+        };
+      }
+  
+      if((squares[1][1].value !== '') && ((squares[1][1].value === squares[0][0].value && squares[2][2].value === squares[0][0].value) || (squares[1][1].value === squares[0][2].value && squares[1][1].value === squares[2][0].value))){
+        return `Winner is player ${squares[1][1].value}`;
+      };
```

Comment:
You had test errors because you didn't match the expected format:

```suggestion
      for(let i in squares){
        if (squares[i][0].value === squares[i][1].value && squares[i][1].value === squares[i][2].value &&  squares[i][0].value !== ''){
          return `Winner is ${squares[i][0].value}`;
        }
        else if(squares[0][i].value === squares[1][i].value && squares[1][i].value === squares[2][i].value &&  squares[0][i].value !== ''){
          return `Winner is ${squares[0][i].value}`;
        };
      }
  
      if((squares[1][1].value !== '') && ((squares[1][1].value === squares[0][0].value && squares[2][2].value === squares[0][0].value) || (squares[1][1].value === squares[0][2].value && squares[1][1].value === squares[2][0].value))){
        return `Winner is ${squares[1][1].value}`;
      };
```